src/charm.py

```diff
@@ -74,6 +74,12 @@ def __init__(self, *args):
         self.framework.observe(self.on.config_changed, self._update_all_metrics_consumers)
         # Ensure we refresh scrape jobs on charm upgrade
         self.framework.observe(self.on.upgrade_charm, self._update_all_metrics_consumers)
+        # Initial charm setup
+        self.framework.observe(self.on.install, self._on_install)
```

Comment:
Note that on upgrade, the `install` event is not emitted.
Not a big deal for this charm.

Reply:
Yeah that is fine as the version will not change.